data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/RequestHandler.java

```diff
@@ -179,8 +183,13 @@ public void handle(final HttpServerRequest request) {
   public Future<Void> onDeleteIngress(
     DataPlaneContract.Resource resource,
     DataPlaneContract.Ingress ingress) {
-    var producer = this.producers.get().remove(ingress.getPath())
-      .getValue().producer;
+    //TODO(slinkydeveloper) https://github.com/knative-sandbox/eventing-kafka-broker/issues/332
+    AtomicReference<Producer<K, V>> producerAtomicReference = new AtomicReference<>();
```

Comment:
Does `producerAtomicReference` need to be atomic?

Reply:
Yep, https://www.baeldung.com/java-lambda-effectively-final-local-variables I prefer to use AtomicReference more than the array for the holder, but it's just a personal preference

Reply:
Hopefully we'll remove this anyway

Reply:
> Hopefully we'll remove this anyway

:+1: